step/step.go

```diff
@@ -0,0 +1,190 @@
+package step
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/bitrise-io/go-steputils/v2/stepconf"
+	"github.com/bitrise-io/go-utils/v2/env"
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/bitrise-steplib/bitrise-step-xcode-test-without-building/xcodebuild"
+	"github.com/kballard/go-shellquote"
+)
+
+const (
+	testResultBundleKey       = "BITRISE_XCRESULT_PATH"
+	zippedTestResultBundleKey = "BITRISE_XCRESULT_ZIP_PATH"
+)
+
+const (
+	timeOutMessageIPhoneSimulator            = "iPhoneSimulator: Timed out waiting"
+	timeOutMessageUITest                     = "Terminating app due to uncaught exception '_XCTestCaseInterruptionException'"
+	earlyUnexpectedExit                      = "Early unexpected exit, operation never finished bootstrapping - no restart will be attempted"
+	failureAttemptingToLaunch                = "Assertion Failure: <unknown>:0: UI Testing Failure - Failure attempting to launch <XCUIApplicationImpl:"
+	failedToBackgroundTestRunner             = `Error Domain=IDETestOperationsObserverErrorDomain Code=12 "Failed to background test runner.`
+	appStateIsStillNotRunning                = `App state is still not running active, state = XCApplicationStateNotRunning`
+	appAccessibilityIsNotLoaded              = `UI Testing Failure - App accessibility isn't loaded`
+	testRunnerFailedToInitializeForUITesting = `Test runner failed to initialize for UI testing`
+	timedOutRegisteringForTestingEvent       = `Timed out registering for testing event accessibility notifications`
+	testRunnerNeverBeganExecuting            = `Test runner never began executing tests after launching.`
+	failedToOpenTestRunner                   = `Error Domain=FBSOpenApplicationServiceErrorDomain Code=1 "The request to open.*NSLocalizedFailureReason=The request was denied by service delegate \(SBMainWorkspace\)\.`
+)
```

Comment:
These appear exactly as they are in the xcode-test step. And also we discussed that you made a great improvement in how we are handling the known xcodebuild failures in this step.

It feels like that code could be extracted and shared between these two test related steps. It would be great to create a ticket somewhere as a reminder. (If there is not one already.)

Reply:
Task created.